lib/json_api_controller/check_resources_exist.rb

```diff
@@ -0,0 +1,33 @@
+module JsonApiController
+  class AccessDenied < StandardError; end
+
+  module CheckResourcesExist
+    extend ActiveSupport::Concern
+
+    included do
+      before_action :check_controller_resources, except: :create
```

Comment:
Long term i think these should be explicitly defined in each controller to make it obvious how the authorization schemes work. I started doing this in #2696 e.g. the ClassificationsController was a first crack at how this might look.

Reply:
Cool, I'll have a look